src/main/java/net/sf/jabref/logic/importer/fileformat/MrDLibImporter.java

```diff
@@ -0,0 +1,261 @@
+    private final String DESCRIPTION = "Takes valid xml documents. Parses from MrDLib API a BibEntriy";
+    private static final Log LOGGER = LogFactory.getLog(MrDLibImporter.class);
+
+
+    /**
+     *
+     * @see net.sf.jabref.logic.importer.Importer#isRecognizedFormat(java.io.BufferedReader)
+     */
+    @Override
+    public boolean isRecognizedFormat(BufferedReader input) throws IOException {
+        String recommendationsAsString = convertToString(input);
+        // check for valid format
+        try {
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            SAXParser saxParser = factory.newSAXParser();
+            DefaultHandler handler = new DefaultHandler() {
+                // No Processing here. Just check for valid xml.
+            };
+
+            try (InputStream stream = new ByteArrayInputStream(recommendationsAsString.toString().getBytes())) {
+                saxParser.parse(stream, handler);
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage(), e);
+                return false;
+            }
+        } catch (ParserConfigurationException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        } catch (SAXException e) {
+            LOGGER.error(e.getMessage(), e);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * The SaxParser needs this String. So I convert it here.
+     * @param Takes a BufferedReader with a reference to the XML document delivered by mdl server.
+     * @return Returns an String containing the XML file.
+     * @throws IOException
+     */
+    private String convertToString(BufferedReader input) throws IOException {
+        String line;
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            while ((line = input.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+        return stringBuilder.toString();
+
+    }
+
+    /*
+     * @see net.sf.jabref.logic.importer.Importer#importDatabase(java.io.BufferedReader)
+     */
+    @Override
+    public ParserResult importDatabase(BufferedReader input) throws IOException {
+        // The Bibdatabase that gets returned in the ParserResult.
+        BibDatabase bibDatabase = new BibDatabase();
+        // The document to parse
+        String recommendations = convertToString(input);
+        // The list ob BibEntries with its associated rank
+        List<RankedBibEntry> rankedBibEntries = new ArrayList<>();
+        // The sorted BibEntries gets stored here later
+        List<BibEntry> bibEntries = new ArrayList<>();
+        //Parsing the response with a SAX parser
+        try {
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            SAXParser saxParser = factory.newSAXParser();
+            DefaultHandler handler = new DefaultHandler() {
```

Comment:
New inner Class?
Do you mean to create a Class at the very bottom, or extract it completely in a new class.